crates/driver/src/tests/cases/protocol_fees.rs

```diff
@@ -0,0 +1,143 @@
+use crate::{
+    domain::competition::order,
+    tests::{
+        self,
+        setup::{
+            ab_adjusted_pool,
+            ab_liquidity_quote,
+            ab_order,
+            ab_solution,
+            ExpectedOrder,
+            FeePolicy,
+            Test,
+        },
+    },
+};
+
+async fn protocol_fee_test_case(
+    side: order::Side,
+    fee_policy: FeePolicy,
+    expected_order: ExpectedOrder,
+) {
+    let test_name = format!("Protocol Fee: {side:?} {fee_policy:?}");
+    let quote = ab_liquidity_quote()
+        .sell_amount(50000000000000000000u128.into())
+        .buy_amount(40000000000000000000u128.into());
+    let pool = ab_adjusted_pool(quote);
+    let order = ab_order()
+        .kind(order::Kind::Limit)
+        .sell_amount(50000000000000000000u128.into())
```

Comment:
What's the limit buy amount of the order? I'd slightly prefer if these values were defined in each test case, we don't care too much about receptive code in tests, the most important thing is that a future maintainer can quickly understand the assumptions this test is making when it breaks.